frontend/src/components/elements/PlotlyChart/CustomTheme.tsx

```diff
@@ -0,0 +1,640 @@
+      title: {
+        font: {
+          color: getGray70(theme),
+          size: fontSizes.smPx,
+        },
+        standoff: 12,
+      },
+      minor: {
+        gridcolor: getGray30(theme),
+      },
+      zeroline: false,
+      automargin: true,
+    },
+    margin: {
+      pad: 16,
+      r: 0,
+      l: 0,
+    },
+    hoverlabel: {
+      bgcolor: colors.bgColor,
+      bordercolor: colors.fadedText10,
+      font: {
+        color: getGray70(theme),
+        family: genericFonts.bodyFont,
+        size: fontSizes.twoSmPx,
+      },
+    },
+    coloraxis: {
+      colorbar: {
+        thickness: 16,
+        xpad: 24,
+        ticklabelposition: "outside",
+        outlinecolor: colors.transparent,
+        outlinewidth: 8,
+        len: 0.75,
+        y: 0.5745,
+        title: {
+          font: {
+            color: getGray70(theme),
+            size: fontSizes.smPx,
+          },
+        },
+        tickfont: {
+          color: getGray70(theme),
+          size: fontSizes.twoSmPx,
+        },
+      },
+      colorscale: {
+        ...(hasLightBackgroundColor(theme)
+          ? {
+              diverging: divergingColorscaleLightTheme,
+              sequential: sequentialColorscaleLightTheme,
+              sequentialminus: sequentialColorscaleDarkTheme,
+            }
+          : {
+              diverging: divergingColorscaleDarkTheme,
+              sequential: sequentialColorscaleDarkTheme,
+              sequentialminus: sequentialColorscaleLightTheme,
+            }),
+      },
+    },
+    colorscale: {
+      ...(hasLightBackgroundColor(theme)
+        ? {
+            diverging: divergingColorscaleLightTheme,
+            sequential: sequentialColorscaleLightTheme,
+            sequentialminus: sequentialColorscaleDarkTheme,
+          }
+        : {
+            diverging: divergingColorscaleDarkTheme,
+            sequential: sequentialColorscaleDarkTheme,
+            sequentialminus: sequentialColorscaleLightTheme,
+          }),
+    },
+    // specifically for the ternary graph
+    ternary: {
+      gridcolor: getGray70(theme),
+      bgcolor: colors.bgColor,
+      title: {
+        font: {
+          family: genericFonts.bodyFont,
+          size: fontSizes.smPx,
+        },
+      },
+      color: getGray70(theme),
+      aaxis: {
+        gridcolor: getGray70(theme),
+        linecolor: getGray70(theme),
+        tickfont: {
+          family: genericFonts.bodyFont,
+          size: fontSizes.twoSmPx,
+        },
+      },
+      baxis: {
+        linecolor: getGray70(theme),
+        gridcolor: getGray70(theme),
+        tickfont: {
+          family: genericFonts.bodyFont,
+          size: fontSizes.twoSmPx,
+        },
+      },
+      caxis: {
+        linecolor: getGray70(theme),
+        gridcolor: getGray70(theme),
+        tickfont: {
+          family: genericFonts.bodyFont,
+          size: fontSizes.twoSmPx,
+        },
+      },
+    },
+  }
+
+  assign(layout, streamlitTheme)
```

Comment:
Assign is used here since we want to fully replace the template and not merge anything, or?

Reply:
I used assign because Plotly has those definitions applied within the javascript object and so if we use a merge, that would not really apply any changes.

Reply:
the problem with just `merge` is that it does not work for arrays. Thats why I used it this way in vega lite: 

```
  return mergeWith({}, streamlitTheme, config, (_, b) =>
    isArray(b) ? b : undefined
  )
``` 
but in the case of plotly, we probably want to replace the full template object. Mayeb even just directly overwriting the layout would be sufficient:  `spec.layout = streamlitTheme`

Reply:
But I assume `assign` is fine here 👍